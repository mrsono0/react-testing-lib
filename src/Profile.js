import React from "react";

const Profile = ({username, name}) => {
    return (
        <div>
        <b>{username}1</b> <span>({name})</span>
        </div>
    );
};

export default Profile;