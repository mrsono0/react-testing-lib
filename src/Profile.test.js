import React from "react";
import { render } from "@testing-library/react";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile username="mrsono0" name="손정현" />);
    expect(utils.container).toMatchSnapshot();
  });
  it("shows the props correctly", () => {
    const utils = render(<Profile username="mrsono0" name="손정현" />);
    utils.getByText("mrsono01");
    utils.getByText("(손정현)");
    // utils.getByText("mrsono0 (손정현)");
  });
});