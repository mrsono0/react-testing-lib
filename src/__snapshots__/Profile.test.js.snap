// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      mrsono0
      1
    </b>
     
    <span>
      (
      손정현
      )
    </span>
  </div>
</div>
`;
